homeassistant/components/conversation/default_agent.py

```diff
@@ -592,6 +610,80 @@ def _get_error_text(
         response_str = lang_intents.error_responses.get(response_key)
         return response_str or _DEFAULT_ERROR_TEXT
 
+    def register_trigger(
+        self,
+        sentences: list[str],
+        callback: core.CALLBACK_TYPE,
+        response: str | None = None,
+    ) -> core.CALLBACK_TYPE:
+        """Register a list of sentences that will trigger a callback when recognized."""
+        action = TriggerAction(callback, response=response)
+        with self._trigger_lock:
```

Comment:
You cannot use threading locks inside a callback executed inside the asyncio loop

Reply:
You can just remove the lock, as the asyncio event loop only executes one piece of Python at the same time.